src/ir/effects.h

```diff
@@ -470,7 +484,12 @@ struct EffectAnalyzer
   }
   void visitNop(Nop* curr) {}
   void visitUnreachable(Unreachable* curr) { branchesOut = true; }
-  void visitPop(Pop* curr) { calls = true; }
+  void visitPop(Pop* curr) {
+    // When EH is enabled, any call can throw.
+    if (features.hasExceptionHandling() && catchDepth == 0) {
```

Comment:
why check the feature here - if a pop exists, doesn't that imply we must have exceptions enabled?

Reply:
Hmm, yeah, right. Deleted it.